### PR TITLE
Move WindowComponent to ECS

### PR DESCRIPTION
**Pull request by Macoron**
## About the PR 
Move window component to ECS.

Also make more general `ExaminableDamageComponent` to show entity damage severity when entity is examined. Maybe can be reused for other entities, like airlocks, walls or mobs.

**Screenshots**

**Changelog**

### CODE REVIEW

File: Content.Server/Window/WindowSystem.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        if (args.Handled)
+            return;
+
+        if (component.RateLimitedKnocking && _gameTiming.CurTime < component.LastKnockTime + component.KnockDelay)
+            return;
+
+        SoundSystem.Play(Filter.Pvs(args.Target), component.KnockSound.GetSound(),
+            Transform(args.Target).Coordinates, AudioHelpers.WithVariation(0.05f));
+
+        args.Target.PopupMessageEveryone(Loc.GetString("comp-window-knock"));
```
Comment (metalgearsloth) [inline Content.Server/Window/WindowSystem.cs:34]: Does PopupSystem have an equivalent?

Also in the petting issue we talked about making a generic thing so petting and knocking are similar but this should be fine for now.

File: Content.Server/Window/WindowComponent.cs [hunk 2]
```diff
@@ -1,116 +1,28 @@
...
-
         [DataField("knockDelay")]
         [ViewVariables(VVAccess.ReadWrite)]
-        private TimeSpan _knockDelay = TimeSpan.FromSeconds(0.5);
+        public TimeSpan KnockDelay = TimeSpan.FromSeconds(0.5);
 
         [DataField("rateLimitedKnocking")]
-        [ViewVariables(VVAccess.ReadWrite)] private bool _rateLimitedKnocking = true;
+        [ViewVariables(VVAccess.ReadWrite)]
+        public bool RateLimitedKnocking = true;
```
Comment (metalgearsloth) [inline Content.Server/Window/WindowComponent.cs:20]: Can probably just remove this as having a 0 delay is equivalent.

File: Content.Server/Damage/Systems/ExaminableDamageSystem.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            return;
+
+        var messages = component.MessagesProto.Messages;
+        if (messages.Length == 0)
+            return;
+
+        var level = GetDamageLevel(uid, component);
+        var msg = Loc.GetString(messages[level]);
+        args.PushMarkup(msg);
+    }
```
Comment (metalgearsloth) [inline Content.Server/Damage/Systems/ExaminableDamageSystem.cs:46]: If damage is predicted could this be on the client instead? Just saves the bandwidth.

File: Content.Server/Damage/Systems/ExaminableDamageSystem.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    private int GetDamageLevel(EntityUid uid, ExaminableDamageComponent? component = null,
+        DamageableComponent? damageable = null, DestructibleComponent? destructible = null)
+    {
+        if (!Resolve(uid, ref component, ref damageable, ref destructible))
+            return 0;
+
+        if (component.MessagesProto == null)
+            return  0;
```
Comment (metalgearsloth) [inline Content.Server/Damage/Systems/ExaminableDamageSystem.cs:55]:
```suggestion
            return 0;
```

**Review by metalgearsloth — APPROVED** [review]

File: Content.Server/Damage/Systems/ExaminableDamageSystem.cs (hunk 3 above)
Comment (Macoron) [reply Content.Server/Damage/Systems/ExaminableDamageSystem.cs:46]: Well, damageable is predicted, but `DestructibleComponent` isn't...

**Pull request merged**